internal/pkg/cosign/tsa/client/client.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"bytes"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/digitorus/timestamp"
+	"github.com/pkg/errors"
+)
+
+// TimestampAuthorityClient provides a client to query RFC 3161 timestamp authorities
+type TimestampAuthorityClient struct {
+	Timestamp TimestampAuthorityService
```

Comment:
You don't need this here - the `TimestampAuthorityClient` becomes an impl of that interface as soon as it implements the interface.

Reply:
With the current design, I need a named interface, otherwise I can't inject in a mocked interface. But thinking more, this design is overly complicated, instead I'll just make `TimestampAuthorityClient` an interface and have the mock implement `GetTimestampResponse` and add a method for `GetCertChain`.

Reply:
Reworked, it looks a lot cleaner now (and we dont expose CertChain for the real implementation)